whois-nrtm4/src/main/java/net/ripe/db/nrtm4/NrtmFileStore.java

```diff
@@ -0,0 +1,57 @@
+package net.ripe.db.nrtm4;
+
+import net.ripe.db.nrtm4.util.NrtmFileUtil;
+import org.mariadb.jdbc.internal.logging.Logger;
+import org.mariadb.jdbc.internal.logging.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+
+@Service
+public class NrtmFileStore {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NrtmFileService.class);
+    private static final int OUTPUT_BUFFER_SIZE = 1024 * 1024;
+
+    private final String path;
+
+    NrtmFileStore(
+        @Value("${nrtm.file.path:/tmp}") final String path
```

Comment:
Is it better to just fail if `nrtm.file.path` is undefined? The `/tmp/` directory is typically only 1 GB in size. 

For testing, you can define the property and specify it's cleaned up on exit:
```
        final Path tempDirectory = Files.createTempDirectory("nrtmv4");
        tempDirectory.toFile().deleteOnExit();
        System.setProperty("nrtm.file.path", tempDirectory.toFile().getAbsolutePath());
```